### ISSUE TITLE
don't stop on error

### ISSUE THREAD

**Issue by tobwen** [jqlang/jq#1942]
Let's say `1.json` is bad and `2.json` is fine. When running `jq -e '.' 1.json 2.json` or `jq -e '.' *.json`, _jq_ directly breaks after `1.json` without telling, which filename was broken.

It would be nice to have `jq` keep on processing and even tell, which `input_filename`  was broken.

**Comment by pkoppstein** [comment]
Yes, the error message should include the input_filename.  Meanwhile, there are some workarounds that *might* apply:

1) For input in which each JSON entity is expected to be on a single line (as is the case with JSONLines), you can use this model:

    jq -R 'try fromjson catch input_filename' *.json

2) The jq startup costs are usually negligible, so if localizing a parsing error is important,
one can usually use a shell "for" loop, e.g. 

    for f in *.json ; do jq -e FILTER "$f" || echo "whoops: $f" >&2 ; done

**Comment by tobwen** [comment]
Thanks for your reply. 

>    `jq -R 'try fromjson catch input_filename' *.json`

Tried it this way because of default limits of _Bash_ (Argument list too long):
`find . -type f -print0 | xargs -0 -P40 jq -R 'try fromjson catch input_filename'`

Leads me to 
```
jq: error: cannot allocate memory
xargs: jq: terminated by signal 6
````

Seems like it slurps all the files up (I've got 60 GiB of free memory). So let's lower it:
`find . -type f -print0 | xargs -0 -P40 -n200 jq -R 'try fromjson catch input_filename'`

Yeah, it returns some files, BUT those are false positives! 
```
$ jq -e '.' "false_positive.json" 2>&1 >/dev/null; echo $?
0

$ jq -R "false_positive.json"
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
./false_positive.json
jq: 1 compile error
```
I had this before with setting `-R`... it's not suitable for all JSON files. Maybe this needs a separate bug report.

> 2. More generally, you can use `try ... catch` like this:
>    jq 'def go: try inputs catch (input_filename, go); go'  *.json

Maybe I'm doing it wrong, it outputs the content, not the filename :(

> 3. The jq startup costs are usually negligible, so if localizing a parsing error is important,
>    one can usually use a shell "for" loop, e.g.
>    for f in *.json ; do jq -e FILTER "$f" || echo "whoops: $f" >&2 ; done

True, the startup costs are fine (other than other tools). That's exactky how I am doing it right now! But it's painful, since we're in the "damn much mulit-core CPU" age of time. So I've wrapped it in a function and I'm using `parallel` or `xargs` to run this function in parallel.

**Comment by pkoppstein** [comment]
@tobwen - Your most recent post seems to stray far from the original question, which dealt with just two files that were to be processed sequentially. Please also note that the original posting in this thread was a bit ambiguous in that "keep on processing" can be interpreted in many ways, e.g. "keep on processing the file in which a parsing error was first found" vs "keep on processing by skipping to the next file". Anyway, I think you'll find that if you try to understand the various strategies that I suggested, you'll find that some of them may be more relevant than you think.
